forc-plugins/forc-client/src/util/pkg.rs

```diff
@@ -1,8 +1,142 @@
+authors = ["Fuel Labs <contact@fuel.sh>"]
+entry = "main.sw"
+license = "Apache-2.0"
+name = "proxy_contract"
+
+[dependencies]
+standards = { git = "https://github.com/FuelLabs/sway-standards/", tag = "v0.5.0" }
+"#;
+
+/// Generates source code for proxy contract owner set to the given 'addr'.
+pub(crate) fn generate_proxy_contract_src(addr: &str, impl_contract_id: &str) -> String {
+    format!(
+        r#"
+contract;
+
+use std::execution::run_external;
+use standards::src5::{{AccessError, SRC5, State}};
+use standards::src14::SRC14;
+
+/// The owner of this contract at deployment.
+const INITIAL_OWNER: Identity = Identity::Address(Address::from({addr}));
+
+// use sha256("storage_SRC14") as base to avoid collisions
+#[namespace(SRC14)]
+storage {{
+    // target is at sha256("storage_SRC14_0")
+    target: ContractId = ContractId::from({impl_contract_id}),
+    owner: State = State::Initialized(INITIAL_OWNER),
+}}
+
+impl SRC5 for Contract {{
+    #[storage(read)]
+    fn owner() -> State {{
+        storage.owner.read()
+    }}
+}}
+
+impl SRC14 for Contract {{
+    #[storage(write)]
+    fn set_proxy_target(new_target: ContractId) {{
+        only_owner();
+        storage.target.write(new_target);
+    }}
+}}
+
+#[fallback]
+#[storage(read)]
+fn fallback() {{
+    // pass through any other method call to the target
+    run_external(storage.target.read())
+}}
+
+#[storage(read)]
+fn only_owner() {{
+    require(
+        storage
+            .owner
+            .read() == State::Initialized(msg_sender().unwrap()),
+        AccessError::NotOwner,
+    );
+}}
+"#
+    )
+}
+
+/// Updates the given package manifest file such that the address field under the proxy table updated to the given value.
+/// Updated manifest file is written back to the same location, without thouching anything else such as comments etc.
+/// A safety check is done to ensure the proxy table exists before attempting to udpdate the value.
+pub(crate) fn update_proxy_address_in_manifest(
+    address: &str,
+    manifest: &PackageManifestFile,
+) -> Result<()> {
+    let mut toml = String::new();
+    let mut file = File::open(manifest.path())?;
+    file.read_to_string(&mut toml)?;
+    let mut manifest_toml = toml.parse::<toml_edit::Document>()?;
+    if manifest.proxy().is_some() {
+        manifest_toml["proxy"]["address"] = toml_edit::value(address);
+        let mut file = std::fs::OpenOptions::new()
+            .write(true)
+            .truncate(true)
+            .open(manifest.path())?;
+        file.write_all(manifest_toml.to_string().as_bytes())?;
+    }
+    Ok(())
+}
+
+/// Creates a proxy contract project at the given path, adds a forc.toml and source file.
+pub(crate) fn create_proxy_contract(
+    addr: &str,
+    impl_contract_id: &str,
+    pkg_name: &str,
+) -> Result<PathBuf> {
+    // Create the proxy contract folder.
+    let proxy_contract_dir = user_forc_directory()
+        .join(PROXY_CONTRACT_FOLDER_NAME)
+        .join(pkg_name);
```

Comment:
This generated contract directory stores the ABI that is needed to updated the proxy contract, right?

If so, does it need a unique name, since other contracts could have the same package name?

Reply:
We can also add a timestamp based random id to this, but abi for all contracts are going to be the same, every time this command is run the abi is re-generated. So overriding it is not a problem in terms of functionality of the added feature. The reason i am not deleting this after deployment is just to keep a housekeeping data in the disk to troubleshoot, we can add the random identifier to make sure we have separate entries for each deployment.